internal/providers/terraform/azure/api_management.go

```diff
@@ -0,0 +1,118 @@
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMApiManagementRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_api_management",
+		RFunc: NewAzureRMApiManagement,
+		ReferenceAttributes: []string{
+			"certificate_id",
+		},
+	}
+}
+
+func NewAzureRMApiManagement(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	costComponents := []*schema.CostComponent{}
+	location := d.Get("location").String()
+	skuName := d.Get("sku_name").String()
+	var tier string
+	var capacity decimal.Decimal
+	if s := strings.Split(skuName, "_"); len(s) == 2 {
+		tier = s[0]
+		capacity, _ = decimal.NewFromString(s[1])
+	}
+
+	if strings.ToLower(tier) != "consumption" {
+		costComponents = append(costComponents, apiManagementCostComponent(
+			fmt.Sprintf("API management (%s)", tier),
+			"units",
+			location,
+			tier,
+			&capacity))
+
+	} else {
+		var apiCalls *decimal.Decimal
+		if u != nil && u.Get("monthly_api_calls").Type != gjson.Null {
+			apiCalls = decimalPtr(decimal.NewFromInt(u.Get("monthly_api_calls").Int()))
+		}
+
+		if apiCalls != nil {
+			if apiCalls.GreaterThan(decimal.NewFromInt(1_000_000)) {
+				apiCalls = decimalPtr(apiCalls.Sub(decimal.NewFromInt(1_000_000)).Div(decimal.NewFromInt(10000)))
+				costComponents = append(costComponents, consumptionAPICostComponent(location, tier, apiCalls))
+			}
+		} else {
+			costComponents = append(costComponents, consumptionAPICostComponent(location, tier, apiCalls))
+		}
+	}
+
+	if strings.ToLower(tier) == "premium" {
+		var selfHostedGateways *decimal.Decimal
+		if u != nil && u.Get("self_hosted_gateway_count").Type != gjson.Null {
+			selfHostedGateways = decimalPtr(decimal.NewFromInt(u.Get("self_hosted_gateway_count").Int()))
+		}
+		costComponents = append(costComponents, apiManagementCostComponent(
+			"Self hosted gateway",
+			"gateways",
+			location,
+			"Gateway",
+			selfHostedGateways,
+		))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func apiManagementCostComponent(name, unit, location, tier string, quantity *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:           name,
+		Unit:           unit,
+		UnitMultiplier: schema.HourToMonthUnitMultiplier,
+		HourlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("azure"),
+			Region:        strPtr(location),
+			Service:       strPtr("API Management"),
+			ProductFamily: strPtr("Developer Tools"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "skuName", ValueRegex: strPtr(fmt.Sprintf("/^%s$/i", tier))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+
+func consumptionAPICostComponent(location, tier string, quantity *decimal.Decimal) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:            "API management (consumption over 1M)",
```

Comment:
https://azure.microsoft.com/en-us/pricing/details/api-management/ mentions that "Monthly free grant of 1 million requests per Azure subscription".

We've been ignoring the per-account free tiers as the discounts are usually tiny, they slightly complicate the logic/output, and it's not possible for us to tell if this TF project will benefit from them or not. So maybe we can do the same here and remove the `(consumption over 1M)` and the logic above that subtracts 1M?